### PR TITLE
DAOS-13559 vos: MD-on-SSD phase2 landing

### PR DESCRIPTION
**Pull request by NiuYawei**
Landing MD-on-SSD phase2 branch to master.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>
Signed-off-by: Sherin T George <sherin-t.george@hpe.com>
Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by tanabarr — DISMISSED** [review]

**Comment by NiuYawei** [comment]
@tanabarr any thoughts on the test failures? The two NLT memcheck failures looks are related to Go, are they known issue? The other two list_verbose.py failures are related to the new 'mem_file_bytes', will you fix that after this landed?

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Comment by tanabarr** [comment]
> @tanabarr any thoughts on the test failures? The two NLT memcheck failures looks are related to Go, are they known issue? The other two list_verbose.py failures are related to the new 'mem_file_bytes', will you fix that after this landed?

https://github.com/daos-stack/daos/pull/15405 addresses the `list_verbose.py` failures so if we have to re-push let's include that otherwise it can land to master after. I'm looking into the memcheck failures.

File: debian/changelog
[the remark this answers is not in the document]
Comment (tanabarr) [reply debian/changelog:1]: "file file"? could you please elaborate

[the remark this answers is not in the document]
Comment (brianjmurrell) [reply debian/changelog:1]: Comment updated.

**Review by jolivier23 — DISMISSED** [review]

**Comment by jolivier23** [comment]
> @tanabarr any thoughts on the test failures? The two NLT memcheck failures looks are related to Go, are they known issue? The other two list_verbose.py failures are related to the new 'mem_file_bytes', will you fix that after this landed?

Do you have the memcheck suppressions from master branch?   The race_amd64.s usually are handled by adding a new suppression but I'm surprised it would be different or MD on SSD branch.

File: debian/changelog
[the remark this answers is not in the document]
Comment (NiuYawei) [reply debian/changelog:1]: Isn't the following entry sufficient? (see the "Add DAV v2 lib" below).

File: utils/rpms/daos.spec
[the remark this answers is not in the document]
Comment (brianjmurrell) [reply utils/rpms/daos.spec:1]: Yes, both `%changelog` and `debian/changelog` entries should be at the top and should bump the `Release:` value in the `daos.spec` and the respective _version-release_ in the `debian/changelog` to +1 from the previous entry.  The existing entry in both of those files in this PR is actually duplicating/conflicting with an already used _version-release_: `2.5.100-10`.

The _version-release_ that should be used for this PR is `2.7.100-10`.

[the remark this answers is not in the document]
Comment (NiuYawei) [reply utils/rpms/daos.spec:1]: Thanks! changelogs are updated.

**Comment by NiuYawei** [comment]
> > @tanabarr any thoughts on the test failures? The two NLT memcheck failures looks are related to Go, are they known issue? The other two list_verbose.py failures are related to the new 'mem_file_bytes', will you fix that after this landed?
> 
> #15405 addresses the `list_verbose.py` failures so if we have to re-push let's include that otherwise it can land to master after. I'm looking into the memcheck failures.

I included your fix from #15405. Thanks!

**Comment by NiuYawei** [comment]
> > @tanabarr any thoughts on the test failures? The two NLT memcheck failures looks are related to Go, are they known issue? The other two list_verbose.py failures are related to the new 'mem_file_bytes', will you fix that after this landed?
> 
> Do you have the memcheck suppressions from master branch? The race_amd64.s usually are handled by adding a new suppression but I'm surprised it would be different or MD on SSD branch.

I didn't add suppression since I'm not quite sure if they are caused by the new changes in control plane. Maybe we can fix them after this landed.

**Review by tanabarr — DISMISSED** [review]

**Comment by tanabarr** [comment]
@brianjmurrell do the build related changes look okay?

File: debian/changelog [hunk 1]
```diff
@@ -135,7 +142,7 @@ daos (2.5.100-12) unstable; urgency=medium
 
  -- Jerome Soumagne <jerome.soumagne@intel.com>  Wed, 15 Nov 2023 10:30:00 -0600
 
-daos (2.5.100-10) unstable; urgency=medium
+daos (2.5.100-11) unstable; urgency=medium
```
Comment (phender) [inline debian/changelog:145]: This should be reverted
```suggestion
daos (2.5.100-10) unstable; urgency=medium
```

**Review by phender — CHANGES_REQUESTED** [review]

**Comment by tanabarr** [comment]
> > @tanabarr any thoughts on the test failures? The two NLT memcheck failures looks are related to Go, are they known issue? The other two list_verbose.py failures are related to the new 'mem_file_bytes', will you fix that after this landed?
> 
> Do you have the memcheck suppressions from master branch? The race_amd64.s usually are handled by adding a new suppression but I'm surprised it would be different or MD on SSD branch.

yes the suppression 's file seems to be the same on both branches

**Comment by tanabarr** [comment]
@NiuYawei @gnailzenh I've pushed again with the following fixes because the previously CI run failed to run the hardware stages because of Jenkins issues.

- Fixed changelog merge mistake identified by @phender 
- Updated test tags to
   Test-tag: pr daily_regression hw,medium,ListVerboseTest
    Skip-func-hw-test-medium-md-on-ssd: false
    Skip-func-hw-test-large-md-on-ssd: false
    Allow-unstable-test: true
- Removed two unnecessary changes on feature branch

**Review by phender — APPROVED** [review]
Ftest changes look good.

**Review by tanabarr — APPROVED** [review]

**Comment by tanabarr** [comment]
@NiuYawei @gnailzenh @phender 
I've been trying to get coverage over the weekend by rerunning to avoid the intermittent issue that stopping tests from starting. https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-15429/

Following builds all on tip of PR branch without any code changes between.

Build 4:
- Functional Hardware Large DID NOT RUN
- Functional Hardware Large MD-on-SSD DID NOT RUN
- Functional Hardware Medium failed with a single known issue DAOS-16768
- Functional Hardware Medium MD-on-SSD DID NOT RUN
- Functional Hardware Verbs Provider DID NOT RUN

Build 5:
- Functional Hardware Large DID NOT RUN
- Functional Hardware Large MD-on-SSD PASSED
- Functional Hardware Medium MD-on-SSD failed because of faulty hardware in cluster on wolf-217
- Functional Hardware Verbs Provider DID NOT RUN

Build 7:
- Functional Hardware Large ???
- Functional Hardware Medium MD-on-SSD ???
- Functional Hardware Verbs Provider ??? 

Summary:
- Functional Hardware Large ???
- Functional Hardware Large MD-on-SSD PASSED
- Functional Hardware Medium ACCEPTABLE
- Functional Hardware Medium MD-on-SSD ???
- Functional Hardware Verbs Provider ???

I don't think there is huge value in waiting for the verbs provider test results but maybe the other two from build 7?

File: debian/changelog (hunk 1 above)
Comment (tanabarr) [reply debian/changelog]: done

**Review by gnailzenh — APPROVED** [review]

**Pull request merged**